### PR TITLE
Dark Mode first pass

### PR DESCRIPTION
**Pull request by BottleRocket-Colin**
Dark Mode is enabled. It's not pretty but it's functional.  Also added better preview annotations and an example of ParameterProvider for mutliple state objects to simplify code for previews.

### CODE REVIEW

File: compose/src/main/java/com/bottlerocketstudios/compose/resources/Colors.kt [hunk 1]
```diff
@@ -10,19 +11,19 @@ val br_light_red = Color(0xffe66963)
...
 val greyish_brown = Color(0xff515151)
 val white = Color(0xffffffff)
-val background_black = Color(0x05000000)
+val mostly_white = Color(0xfff8f8f8)
 val black = Color(0xff000000)
 val sea_foam = Color(0xff1fada8)
-val brown_grey_two = Color(0xff797979)
-val brown_grey_three = Color(0xff9e9e9e)
+// val brown_grey_two = Color(0xff797979)
+// val brown_grey_three = Color(0xff9e9e9e)
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/resources/Colors.kt]: @BottleRocket-Colin Are these colors used in the zeplin project comps? Do we still need them? If not, remove them. If so, why comment them out?

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryBrowserScreen.kt [hunk 2]
```diff
@@ -179,3 +171,10 @@ private val browserItems = listOf(
         isFolder = false
     ),
 )
+
+val previewState = RepositoryBrowserScreenState(
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryBrowserScreen.kt]: @BottleRocket-Colin nit: Make this `private`

```suggestion
private val previewState = RepositoryBrowserScreenState(
```

File: compose/src/main/java/com/bottlerocketstudios/compose/resources/Theme.kt [hunk 3]
```diff
@@ -36,10 +38,11 @@ private val LocalAppDimens = staticCompositionLocalOf {
 
 @Composable
 fun ArchitectureDemoTheme(
+    darkTheme: Boolean = isSystemInDarkTheme(),
     content: @Composable () -> Unit
 ) {
     // TODO This should be updated later to support a dark mode check
-    val colors = lightColors
+    val colors = if (darkTheme) darkColors else lightColors
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/resources/Theme.kt:45]: Nice!

File: compose/src/main/java/com/bottlerocketstudios/compose/snippets/CreateSnippetScreen.kt [hunk 4]
```diff
@@ -145,38 +144,27 @@ fun LabelledCheckbox(isPrivate: Boolean, onCheckedChange: (Boolean) -> Unit) {
...
-    val isPrivate: State<Boolean>,
-    val creationFailed: State<Boolean>,
-    val createEnabled: State<Boolean>,
-    val onTitleChanged: (String) -> Unit,
-    val onFilenameChanged: (String) -> Unit,
-    val onContentsChanged: (String) -> Unit,
-    val onPrivateChanged: (Boolean) -> Unit,
-    val onCreateClicked: () -> Unit,
+@Composable
+fun previewState() = CreateSnippetScreenState(
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/snippets/CreateSnippetScreen.kt]: nit: Make `private`

```suggestion
private fun previewState() = CreateSnippetScreenState(
```

File: compose/src/main/java/com/bottlerocketstudios/compose/snippets/SnippetsBrowserScreen.kt [hunk 5]
```diff
@@ -66,47 +65,12 @@ private fun SnippetsFabLayout(onFabClick: () -> Unit) {
     )
 }
 
-@Preview(showSystemUi = true)
+@PreviewAll
 @Composable
-fun SnippetsBrowserScreenPreview() {
+fun SnippetsBrowserScreenPreview(
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/snippets/SnippetsBrowserScreen.kt]: nit: Make `private`

```suggestion
private fun SnippetsBrowserScreenPreview(
```

File: compose/src/main/java/com/bottlerocketstudios/compose/snippets/SnippetsPreviewData.kt [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    id = "123456789XYZ",
+    workspaceSlug = "workspace_slug",
+    title = "Private",
+    userName = "Bob Ross",
+    formattedLastUpdatedTime = StringIdHelper.Raw(""),
+)
+
+internal val listOfMockSnippets = listOf(mockSnippet1, mockSnippet1)
+
+class SnippetBrowserPreviewProvider : PreviewParameterProvider<SnippetsBrowserScreenState> {
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/snippets/SnippetsPreviewData.kt]: @BottleRocket-Colin nit: Can this class be made `internal`?

```suggestion
internal class SnippetBrowserPreviewProvider : PreviewParameterProvider<SnippetsBrowserScreenState> {
```

This is interesting!

File: compose/src/main/java/com/bottlerocketstudios/compose/snippets/SnippetsBrowserScreen.kt [hunk 7]
```diff
@@ -66,47 +65,12 @@ private fun SnippetsFabLayout(onFabClick: () -> Unit) {
...
+    @PreviewParameter(SnippetBrowserPreviewProvider::class) state: SnippetsBrowserScreenState
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/snippets/SnippetsBrowserScreen.kt:71]: Very cool!

File: compose/src/main/java/com/bottlerocketstudios/compose/util/PreviewUtil.kt [hunk 8]
```diff
@@ -1,20 +1,35 @@
...
+@PreviewLightDark
+annotation class PreviewAll
+
+@Preview(showSystemUi = true, device = Devices.DEFAULT, group = "Default")
+@Preview(showSystemUi = true, device = Devices.DEFAULT, group = "Default", uiMode = UI_MODE_NIGHT_YES)
+annotation class PreviewLightDark
+
+@Preview(showBackground = true)
+@Preview(showBackground = true, uiMode = UI_MODE_NIGHT_YES)
+annotation class PreviewComposable
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/util/PreviewUtil.kt:35]: Great idea!

File: compose/src/main/java/com/bottlerocketstudios/compose/util/PreviewUtil.kt [hunk 9]
```diff
@@ -1,20 +1,35 @@
...
-@Composable
 fun <T> T.asMutableState() = mutableStateOf(this)
 
-@Preview(showSystemUi = true, device = Devices.FOLDABLE)
-@Preview(showSystemUi = true, device = Devices.TABLET)
-@Preview(showSystemUi = true, device = Devices.DEFAULT)
-annotation class PreviewAllDevices
+@Preview(showSystemUi = true, device = Devices.FOLDABLE, group = "Foldable")
+@Preview(showSystemUi = true, device = Devices.TABLET, group = "Tablet")
+@PreviewLightDark
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/util/PreviewUtil.kt:26]: This is cool how you can nest previews!

**Review by jaredstockton — APPROVED** [review]
A few nits but overall it looks great!

File: compose/src/main/java/com/bottlerocketstudios/compose/resources/Colors.kt (hunk 1 above)
Comment (BottleRocket-Colin) [reply compose/src/main/java/com/bottlerocketstudios/compose/resources/Colors.kt]: Good point... I'm deleting with extreme action!

**Pull request merged**